diff --git a/node_modules/@glenstack/cf-workers-discord-bot/dist/setup.js b/node_modules/@glenstack/cf-workers-discord-bot/dist/setup.js
index d3c8ab0..7b061e6 100644
--- a/node_modules/@glenstack/cf-workers-discord-bot/dist/setup.js
+++ b/node_modules/@glenstack/cf-workers-discord-bot/dist/setup.js
@@ -46,6 +46,16 @@ const deleteExistingCommands = ({ applicationID }, authedFetch) => __awaiter(voi
 });
 const createCommands = ({ applicationID, commands, }, authedFetch) => __awaiter(void 0, void 0, void 0, function* () {
     const url = `https://discord.com/api/v8/applications/${applicationID}/commands`;
+    const getRequest = new Request(url, {
+        method: "GET"
+    });
+    const getResponse = yield authedFetch(getRequest);
+    if (getResponse.ok) {
+        console.log(`Got OK response from ${url}`);
+        console.log(yield getResponse.json());
+    } else {
+        console.error(`Got ${getResponse.status} response from ${url}`);
+    }
     const promises = commands.map(([command, handler]) => __awaiter(void 0, void 0, void 0, function* () {
         const request = new Request(url, {
             method: "POST",
